evap/staff/views.py

```diff
@@ -1386,23 +1388,67 @@ def evaluation_email(request, evaluation_id):
     )
 
 
-def helper_delete_users_from_evaluation(evaluation, operation):
-    if "participants" in operation:
-        deleted_person_count = evaluation.participants.count()
-        deletion_message = _("{} participants were deleted from evaluation {}")
-        evaluation.participants.clear()
+class ImportAction(Enum):
+    COPY = "copy"
+    IMPORT = "import"
+    TEST = "test"
+
+    @classmethod
+    def from_operation(cls, operation: str) -> "ImportAction":
+        if operation.startswith("copy-"):
+            return cls.COPY
+        if operation.startswith("import-"):
+            return cls.IMPORT
+        if operation.startswith("test-"):
+            return cls.TEST
+        raise ValueError(f"Unknown operation: {operation}")
+
+
+def import_or_copy_participants(
+    request,
+    replace: bool,
+    import_action: Literal[ImportAction.COPY, ImportAction.IMPORT],
+    import_type: ImportType,
+    evaluation: Evaluation,
+    copy_form: EvaluationParticipantCopyForm,
+) -> bool:
+    if replace:
+        if import_type == ImportType.PARTICIPANT:
+            deleted_person_count = evaluation.participants.count()
+            deletion_message = _("{} participants were deleted from evaluation {}")
+            evaluation.participants.clear()
+        elif import_type == ImportType.CONTRIBUTOR:
+            deleted_person_count = evaluation.contributions.exclude(contributor=None).count()
+            deletion_message = _("{} contributors were deleted from evaluation {}")
+            evaluation.contributions.exclude(contributor=None).delete()
+    if import_action == ImportAction.IMPORT:
+        file_content = get_import_file_content_or_raise(request.user.id, import_type)
+        importer_log = import_persons_from_file(import_type, evaluation, test_run=False, file_content=file_content)
+        delete_import_file(request.user.id, import_type)
+    elif import_action == ImportAction.COPY:
+        copy_form.evaluation_selection_required = True
+        if not copy_form.is_valid():
+            return False
+        import_evaluation = copy_form.cleaned_data["evaluation"]
+        importer_log = import_persons_from_evaluation(
+            import_type, evaluation, test_run=False, source_evaluation=import_evaluation
+        )
     else:
-        assert "contributors" in operation
-        deleted_person_count = evaluation.contributions.exclude(contributor=None).count()
-        deletion_message = _("{} contributors were deleted from evaluation {}")
-        evaluation.contributions.exclude(contributor=None).delete()
+        assert_never(import_action)
 
-    return deleted_person_count, deletion_message
+    if replace:
+        importer_log.add_success(  # pylint: disable=possibly-used-before-assignment  # see https://www.github.com/pylint-dev/pylint/issues/9780
+            format_html(deletion_message, deleted_person_count, evaluation.full_name),
+            category=ImporterLogEntry.Category.RESULT,
+        )
```

Comment:
I'd expect the `pylint` comment to still mask issues with the possible usage of `deletion_message` etc. before usage.

Pylint has been updated in the meantime, with 3.2.6 I don't get the pylint error here, so I guess we should be fine without the `disable` comment.

Reply:
yep :+1: